compiler/il/OMRNode.cpp

```diff
@@ -3911,13 +3911,25 @@ OMR::Node::createStoresForVar(TR::SymbolReference * &nodeRef, TR::TreeTop *inser
                   }
                else
                   {
-                  newArrayRef = comp->getSymRefTab()->
-                     createTemporary(comp->getMethodSymbol(), TR::Address);
+                  TR::Node *arrayObjectNode = NULL;
+                  TR::Node *arrayLoadNode = NULL;
 
-                  TR::Node *newStore = TR::Node::createStore(newArrayRef, child);
+                  if (child->isDataAddrPointer())
+                     arrayObjectNode = child->getFirstChild();
+
+                  newArrayRef = comp->getSymRefTab()->createTemporary(comp->getMethodSymbol(), TR::Address);
+                  TR::Node *newStore = TR::Node::createStore(newArrayRef, arrayObjectNode ? arrayObjectNode : child);
                   newStoreTree = TR::TreeTop::create(comp, newStore);
+
                   newArrayRef->getSymbol()->setPinningArrayPointer();
                   pinningArray = newArrayRef->getSymbol()->castToAutoSymbol();
+
+                  if (child->isDataAddrPointer())
+                     {
+                     arrayLoadNode = TR::Node::createLoad(arrayObjectNode, newArrayRef);
+                     child->setAndIncChild(0, arrayLoadNode);
```

Comment:
you are changing the grand-child array object load to the pinning-array local load, but i didn't see previous code (for non-offheap case) changing the child into the corresponding pinning array local load.  is it done somewhere else?

I would like to ask @vijaysun-omr for a review too.

Reply:
My intent was take a similar approach to what was done in https://github.com/eclipse/omr/commit/c86da7e3aae883dd226d3a23c1010e01cb4d2dbc:
https://github.com/eclipse/omr/blob/fa23d1b84b93f1b6b199f085b88b53bfa50ccee4/compiler/il/OMRNode.cpp#L3788-L3800

Reply:
I agree with Julian that we probably don't need to change the child to be a load of the newly created temp (the code after line 3927). It probably doesn't hurt doing so either. If you have time to test without the changes to use the load, that may be worth it to minimize what is being changed.